test/test_components.py

```diff
@@ -0,0 +1,211 @@
+import numpy as np
+import pytest
+
+import pypsa
+
+
+@pytest.fixture
+def network():
+    return pypsa.examples.ac_dc_meshed(from_master=True)
+
+
+@pytest.fixture
+def empty_network_5_buses():
+    # Set up empty network with 5 buses.
+    network = pypsa.Network()
+    n_buses = 5
+    for i in range(n_buses):
+        network.add(
+            'Bus',
+            f'bus_{i}'
+        )
+    return network
+
+
+def test_mremove(network):
+    """
+    GIVEN   the AC DC exemplary pypsa network
+
+    WHEN    two components of Generator are removed with mremove
+
+    THEN    the generator dataframe and the time-dependent generator dataframe
+                should not contain the removed elements.
+    """
+    generators = ['Manchester Wind', 'Frankfurt Wind']
+
+    network.mremove('Generator', generators)
+
+    assert generators not in network.generators.index.tolist()
+    assert generators not in network.generators_t.p_max_pu.columns.tolist()
+
+
+def test_mremove_misspelled_component(network, caplog):
+    """
+    GIVEN   the AC DC exemplary pypsa network
+
+    WHEN    a misspelled component is removed with mremove
+
+    THEN    the function should not change anything in the Line component
+                dataframe and an error should be logged.
+    """
+
+    len_lines = len(network.lines.index)
+
+    network.mremove('Liness', ['0', '1'])
+
+    assert len_lines == len(network.lines.index)
+    assert caplog.records[-1].levelname == 'ERROR'
```

Comment:
this caplog behaviour is very interesting!

I'd say to choose one of `test_mremove_misspelled_component` and `test_mremove_non_component` since the tested behaviour is the same

Reply:
True! I removed `test_mremove_non_component()`.